frontend/catalyst/third_party/oqd/oqd_device.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+OQD Device
+~~~~~~~~~~
+
+This module defines the classes that represent an Open Quantum Design (OQD)
+trapped-ion quantum computer device.
+"""
+from typing import Optional
+
+from pennylane.devices import Device, ExecutionConfig
+from pennylane.transforms.core import TransformProgram
+from catalyst.compiler import get_lib_path
+
+BACKENDS = ["default"]
+
+
+class OQDDevice(Device):
+    """The OQD device allows to access the hardware devices from OQD using Catalyst."""
+
+    config_filepath = get_lib_path("oqd_runtime", "OQD_LIB_DIR") + "/backend" + "/oqd.toml"
+
+    def __init__(self, wires, backend, shots=1024, **kwargs):
```

Comment:
I understand that this is a real quantum hardware device so a default shots of None (or zero) does not make sense, since those are interpreted as analytical results. However, does a default shots of 1 make more sense? So that the default cost is one run of the hardware, instead of 1024 runs.

Reply:
I agree, I'll change it to `shots=1`. This PR is still very rough and mostly just boilerplate that I've copied over from the oqc device ;)

Reply:
723e8dee

Reply:
Note that IBM/Qiskit set a standard with their HW, with the default number of shots being 1024. So 1024 is the expectation these days for the default number of shots.

I recommend leaving this at 1024, but we should chat to the OQD team to understand if there is any trapped-ion/HW specific reason to have a different default.

Reply:
Thanks Josh! Reverted back to 1024 in df1d265f.

Reply:
@josh146 I was not aware of this convention, thanks!

Reply:
Why not make it a required argument since it is, and assume any defaults?

Reply:
Also a good option!

Reply:
I agree, I've made it a required argument in 13c60090.